// 0 for nothing, 1 for curb_up, 2 for curb_down
int32 detection